typing/typecore.ml

```diff
@@ -3944,22 +3937,34 @@ and type_label_exp create env loc ty_expected
     let arg = type_argument env sarg ty_arg (instance ty_arg) in
     end_def ();
     try
```

Comment:
You're adding a lot of extra code in this case. Couldn't it be factorized?

Reply:
I've now factorized out the various `generalize` calls.